base/settings/R/check.all.settings.R

```diff
@@ -147,160 +167,180 @@ check.database <- function(database) {
+    PEcAn.logger::logger.severe("Missing migration 20140729045640,",
+      "this introduces modeltypes table")
   }
   if (! ("20151011190026" %in% versions)) {
-    PEcAn.logger::logger.severe("Missing migration 20151011190026, this introduces notes and user_id in workflows")
+    PEcAn.logger::logger.severe("Missing migration 20151011190026,",
+      "this introduces notes and user_id in workflows")
   }
-  
+
   # check if database is newer
-  last_migration_date <- lubridate::ymd_hms(tail(versions, n = 1))
-  pecan_release_date <- lubridate::ymd(utils::packageDescription("PEcAn.DB")$Date)
+  last_migration_date <- lubridate::ymd_hms(utils::tail(versions, n = 1))
+  pecan_release_date <- lubridate::ymd(
+    utils::packageDescription("PEcAn.DB")$Date)
   if (last_migration_date > pecan_release_date) {
     PEcAn.logger::logger.warn(
-      "Last database migration", tail(versions, n = 1),
+      "Last database migration", utils::tail(versions, n = 1),
       "is more recent than this", pecan_release_date, "release of PEcAn.",
       "This could result in PEcAn not working as expected.")
   }
 }
 
-##' Sanity checks. Checks the settings file to make sure expected fields exist. It will try to use
-##' default values for any missing values, or stop the exection if no defaults are possible.
-##'
-##' Expected fields in settings file are:
-##' - pfts with at least one pft defined
-##' @title Check Settings
-##' @param settings settings file
-##' @return will return the updated settings values with defaults set.
-##' @author Rob Kooper, David LeBauer
-##' @export check.settings
-check.settings <- function(settings, force=FALSE) {
-  if(!force && !is.null(settings$settings.info$checked) && settings$settings.info$checked==TRUE) {
+#' Sanity checks. Checks the settings file to make sure expected fields exist.
+#'  It will try to use default values for any missing values,
+#'  or stop the exection if no defaults are possible.
+#'
+#' Expected fields in settings file are:
+#' - pfts with at least one pft defined
+#' @title Check Settings
+#' @param settings settings file
+#' @return will return the updated settings values with defaults set.
+#' @author Rob Kooper, David LeBauer
+#' @export check.settings
+check.settings <- function(settings, force = FALSE) {
+  if (!force
+      && !is.null(settings$settings.info$checked)
+      && settings$settings.info$checked == TRUE) {
     PEcAn.logger::logger.info("Settings have been checked already. Skipping.")
     return(invisible(settings))
   } else {
     PEcAn.logger::logger.info("Checking settings...")
   }
-  
-  if(is.MultiSettings(settings)) {
-    return(invisible(papply(settings, check.settings, force=force)))
+
+  if (is.MultiSettings(settings)) {
+    return(invisible(papply(settings, check.settings, force = force)))
   }
-  
-  scipen = getOption("scipen")
-  on.exit(options(scipen=scipen))
-  options(scipen=12)
-  
+
+  scipen <- getOption("scipen")
+  on.exit(options(scipen = scipen))
```

Comment:
I had not heard of on.exit. We should use this to close database connections if we open them in a function. This will remove the kludge I have right now sometimes, `if open connection set closeit=True` and later `if closeit db.close(con)`.

Reply:
@robkooper on.exit is great, but one pitfall to know: You nearly always want to use it with `add=TRUE`, otherwise it by default it *replaces* any previously requested on.exit operations. I should add that here for defensiveness.